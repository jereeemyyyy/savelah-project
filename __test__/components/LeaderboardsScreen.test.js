import React from 'react';
import renderer from 'react-test-renderer';
import LeaderboardScreen from '../../app/screens/leaderboards/LeaderBoardsScreen';

// Mock LeaderboardList component
jest.mock('../../app/screens/leaderboards/components/LeaderBoardList', () => 'LeaderboardList');

describe('LeaderboardScreen', () => {
    it('renders correctly', () => {
        const tree = renderer.create(<LeaderboardScreen />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('contains the expected text elements', () => {
        const component = renderer.create(<LeaderboardScreen />);
        const instance = component.root;

        expect(instance.findByProps({testID: 'leaderboards'}).props.children).toBe('Leaderboards');
    });

    it('contains the LeaderboardList component', () => {
        const component = renderer.create(<LeaderboardScreen />);
        const instance = component.root;

        expect(instance.findByType('LeaderboardList')).toBeTruthy();
    });
});
