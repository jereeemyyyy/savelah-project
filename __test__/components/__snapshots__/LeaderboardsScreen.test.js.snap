// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LeaderboardScreen renders correctly 1`] = `
<View
  style={
    [
      {
        "flexBasis": "0%",
        "flexGrow": 1,
        "flexShrink": 1,
      },
      {
        "backgroundColor": "#1f2937",
      },
    ]
  }
>
  <Text
    style={
      [
        {
          "fontSize": 30,
          "lineHeight": 36,
        },
        {
          "fontWeight": "700",
        },
        {
          "paddingBottom": 16,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 16,
        },
        {
          "textAlign": "left",
        },
        {
          "color": "#fff",
        },
      ]
    }
    testID="leaderboards"
  >
    Leaderboards
  </Text>
  <LeaderboardList />
</View>
`;
